addon/ember-concurrency.ts

```diff
@@ -0,0 +1,99 @@
+import { task as _task, taskGroup as _taskGroup } from 'ember-concurrency';
+import RSVP from 'rsvp';
+
+export const task = _task as (fn: () => IterableIterator<any>) => TaskProperty;
+
+export const taskGroup = _taskGroup as () => TaskGroupProperty;
+
+interface CommonTaskProperty {
+  restartable: () => TaskProperty;
+  drop: () => TaskProperty;
+  keepLatest: () => TaskProperty;
+  enqueue: () => TaskProperty;
+  maxConcurrency: (n: number) => TaskProperty;
+  cancelOn: (eventName: string) => TaskProperty;
+  group: (groupName: string) => TaskProperty;
+}
+
+export interface TaskProperty extends CommonTaskProperty {
+  evented: () => TaskProperty;
+  debug: () => TaskProperty;
+  on: (eventName: string) => TaskProperty;
+}
+
+export interface TaskGroupProperty extends CommonTaskProperty {}
+
+export type TaskOptions = {
+  [option in keyof TaskProperty]?: Parameters<TaskProperty[option]> extends [
+    infer P
+  ]
+    ? P
+    : true
+};
+export type TaskGroupOptions = {
+  [option in keyof TaskGroupProperty]?: Parameters<
+    TaskGroupProperty[option]
+  > extends [infer P]
+    ? P
+    : true
+};
+
+// Based on https://github.com/CenterForOpenScience/ember-osf-web/blob/7933316efae805e00723789809bdeb58a96a286a/types/ember-concurrency/index.d.ts
+
+export enum TaskInstanceState {
+  Dropped = 'dropped',
+  Canceled = 'canceled',
+  Finished = 'finished',
+  Running = 'running',
+  Waiting = 'waiting'
+}
+
+export interface TaskInstance<T> extends PromiseLike<T> {
+  readonly error?: any;
```

Comment:
Maybe we should try and use `unknown` instead of `any` wherever possible to help prevent folks from accidentally introducing unsafety into their code?

Reply:
I converted some of the `any`s over to `unknown`. I hope I got the right ones. 😅